common/models/wordbreakers/src/data-compiler/index.ts

```diff
@@ -0,0 +1,366 @@
+
+let extendedPictographicCodePoints = readZippedCharacterPropertyFile(emojiDataFilename)
+  .filter(({property}) => property === 'Extended_Pictographic');
+
+// Try generating the regular expression both in a way that is
+// backwards-compatbile and one that only works in ES6+.
+let extendedPictographicRegExp;
+let compatibleRegexp = utf16AlternativesStrategy();
+let es6Regexp = unicodeRangeStrategy();
+
+// Choose the shortest regular expression.
+// In my experience, the ES6 regexp is an order of magnitude smaller!
+if (es6Regexp.length < compatibleRegexp.length) {
+  extendedPictographicRegExp = es6Regexp;
+  console.warn(`Using ES6 regexp [${es6Regexp.length} chars]`);
+} else {
+  extendedPictographicRegExp = compatibleRegexp;
+  console.warn(`Using compatibility regexp [${compatibleRegexp.length} chars]`);
+}
+
+let catIndexSeed = 0;
+const categoryMap = new Map<string, number>();
+
+for(let cat of categories) {
+  categoryMap.set(cat, catIndexSeed++);
+  if(catIndexSeed == '`'.charCodeAt(0)) {
+    catIndexSeed++; // Skip the back-tick as an encoding symbol.
+                    // Reduces complications, as it's the encoding string start/end char.
+  }
+}
+
+const bmpRanges: typeof ranges = [];
+const nonBmpRanges: typeof ranges = [];
+
+// { start: number, property: number}[]
+for(let range of ranges) { // already sorted
+  if(range.start <= 0xFFFF) {
+    bmpRanges.push(range);
+  } else {
+    if(nonBmpRanges.length == 0) {
+      const finalBmpRange = bmpRanges[bmpRanges.length - 1];
+      bmpRanges.push({
+        start: 0xFFFF,
+        property: range.property,
+        end: undefined
+      });
+
+      nonBmpRanges.push({
+        start: 0x10000,
+        property: finalBmpRange.property,
+        end: undefined
+      });
+    }
+
+    nonBmpRanges.push(range);
+  }
+}
+
+//////////////////////// Creating the generated file /////////////////////////
+
+// Save the output in the gen/ directory.
+let stream = fs.createWriteStream(generatedFilename);
+
+// // Former entry in the original version by Eddie that was never included in our repo:
+// export const extendedPictographic = ${extendedPictographicRegExp};
+
+// Generate the file!
+stream.write(`// Automatically generated file. DO NOT MODIFY.
+/**
+ * Valid values for a word break property.
+ */
+export const enum WordBreakProperty {
+${ /* Create enum values for each word break property */
+  Array.from(categories)
+    .map(x => `  ${x} = ${categoryMap.get(x)}`)
+    .join(',\n')
+}
+};
+
+export const WORD_BREAK_PROPERTY_BMP: string = \`${
+  // To consider:  emit `\uxxxx` codes instead of the raw char?
+  bmpRanges.map(({start, property}) => {
+    let codedStart = String.fromCodePoint(start);
+    if(codedStart == '`') {
+      // Prevents accidental unescaped use of the string start/end char.
+      // The backslash gets removed on file-load.
+      codedStart = '\\`';
+    }
+    const codedProp = String.fromCharCode(categoryMap.get(property));
```

Comment:
A possibly-relevant cross-reference from a different draft pred-text PR: https://github.com/keymanapp/keyman/pull/11088#discussion_r1540558119

It may help the encoding to shift the enum values by about 0x20 in order to avoid use of the ASCII control-code range.

Those chars are "fine" for the base character (`codedStart`) since they'll only be used once each, if that, as a key.  Used for _values_, however, the control codes would see very high usage.  Assuming `esbuild` would "escape" control-code characters, frequent use of them would likely greatly lower our file-size savings.